### PR TITLE
feat: enable/disable postgres

### PR DESCRIPTION
**Pull request by xavidop**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by xavidop** [comment]
Implements changes requested on https://github.com/rudderlabs/rudderstack-helm/pull/28

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -188,3 +188,26 @@ Return the appropriate apiVersion for statefulset.
     {{ end }}
 {{- end }}
 {{- end -}}
+
+
+{{/*
+Return the appropriate postgresql main hosts for statefulset.
+*/}}
+{{- define "postgresql.host.main" -}}
```
Comment (bdashrad) [inline templates/_helpers.tpl:196]: this is a nice way to implement this.

**Comment by bdashrad** [comment]
This would be great to have, and fixes #16

File: templates/statefulset.yaml [hunk 2]
```diff
@@ -71,12 +71,14 @@ spec:
...
             value: "true"
           - name: POSTGRES_POD_NAME
-            value: "{{ .Release.Name }}-{{ .Values.postgresql.nameOverride }}"
+            value: "{{ include "postgresql.host.main" . }}"
           - name: POSTGRES_HEADLESS_SVC
-            value: "{{ .Release.Name }}-{{ .Values.postgresql.nameOverride }}-headless"
+            value: "{{ include "postgresql.host.headless" . }}"
+          {{- end }}
           - name: JOBS_DB_HOST
             value: "This is expected to be set by docker entrypoint script"
```
Comment (lokey) [inline templates/statefulset.yaml:83]: `JOBS_DB_HOST` env is how the server connects to the postgres db. This env value is overwritten by the docker entrypoint script in server if `COMPUTE_DB_HOST_IN_K8S`.
When postgres.enabled is false, we should have this env pointing to the external db's host that is used.
We can do this by having an extra config value like `values.backend.exteranlDBHost`.

**Review by lokey — CHANGES_REQUESTED** [review]

**Pull request closed without merging**